tests/integration/cluster_fixture.py

```diff
@@ -0,0 +1,226 @@
+    """
+
+    def __new__(cls, options):
+        conf = {k[6:]: v for (k, v) in os.environ.items() if k.startswith('KAFKA')}
+
+        if bool(conf):
+            return ExternalClusterFixture(conf, options)
+
+        return TrivupFixture(options)
+
+
+class ExternalClusterFixture(object):
+    __slots__ = ['_conf', '_topic']
+
+    """
+        Configures integration tests clients to communicate with external(developer provided) cluster.
+        All client configs are valid and follow the following naming convention.
+    """
+
+    OPTIONAL = ['TOPIC', 'SCHEMA_REGISTRY_URL']
+    REQUIRED = ['BOOTSTRAP_SERVERS']
+    PROPS = OPTIONAL + REQUIRED
+
+    def __init__(self, conf):
+
+        self._topic = conf.pop('TOPIC', "confluent_kafka_test{}".format(str(uuid4())))
+
+        for r in self.REQUIRED:
+            if r not in conf:
+                raise ValueError("{} is a required config".format(conf))
+
+        self._conf = {k.lower().replace('_', '.'): v for (k, v) in conf.items()}
+
+    def client_conf(self, options=None):
+        conf = copy.deepcopy(self._conf)
+
+        if options is not None:
+            conf.update(options)
+
+        return conf
+
+    def producer(self, updates=None):
+        return Producer(self.client_conf(updates))
+
+    def consumer(self, updates):
+        conf = self.client_conf(updates)
+        if 'group.id' not in conf:
+            conf.update({'group.id': uuid1()})
+
+        return Consumer(conf)
+
+    @property
+    def schema_registry(self):
+        return self._conf.get('schema.registry.url')
+
+
+class TrivupFixture(object):
+    __slots__ = ['_cluster', '_admin', '_producer']
+
+    default_options = {
+        'broker_cnt': 1,
+    }
+
+    def __init__(self, options):
+        self._cluster = KafkaCluster(**options)
+        self._admin = None
+        self._producer = None
+        self._cluster.wait_operational()
+
+    def client_conf(self, options=None):
+        conf = self._cluster.client_conf()
+
+        if hasattr(self._cluster, 'sr'):
+            conf['schema.registry.url'] = self._cluster.sr.get('url')
+
+        if options is not None:
+            conf.update(options)
+
+        return conf
+
+    @classmethod
+    def _topic_conf(cls, options=None):
+        conf = {'num_partitions': 1,
+                'replication_factor': 1}
+
+        if options is not None:
+            conf.update(options)
+
+        return conf
+
+    def producer(self, options=None):
+        """
+        Returns a producer bound to this cluster.
+
+        :param dict options: additional producer configuration values to be used.
+        :returns: a new producer instance.
+        :rtype: confluent_kafka.Producer
+        """
+        return Producer(self.client_conf(options))
+
+    def consumer(self, options=None):
+        """
+        Returns a consumer bound to this cluster.
+
+        :param dict options: additional consumer values to be used.
```

Comment:
what is a consumer value?
If this is configuration we should call it conf, 'values' and 'options' introduce new names for something that already exists and has a common name, thus bringing confusion.